assets/javascripts/redmine_s3.js

```diff
@@ -0,0 +1,15 @@
+/*
+ * This script handles thumbnail image missing errors
+ */
+
+$(function() {
```

Comment:
I think this will make s3 attachments less convenient than usual attachments - the usual redmine behavior is to automatically generate them when they are missing.
By the way I don't think that it violates the principle this badly. If it was redmine without s3 it would be like this:
- Make a call to thumbnail action;
- If thumbnail is missing, generate thumbnail and return link; otherwise just return link.
With S3 it is like this:
- Make a call to thumbnail action;
- If generation param is specified, generate thumbnail and return link; otherwise just return link.
All the script does is a call to server to get thumbnail with param. It's not some new class responsibility, everything is the same.